infrastructure/Makefile

```diff
@@ -91,4 +101,55 @@ remove-reach:
 # halt any dockerized devnets, docker kill and rm all images
 	$(REACH_PATH)/reach down
 	$(REACH_PATH)/reach docker-reset
-	rm -r $(REACH_PATH)
\ No newline at end of file
+	rm -r $(REACH_PATH)
+
+## --- Indexer jobs (docker based)--- ##
+
+indexer-docker-up:
+# clone indexer git repo: https://github.com/algorand/indexer.git and pull
+	@[ -d $(INDEXER_PATH) ] || git clone $(INDEXER_REMOTE_REPO) $(INDEXER_PATH)
+	(cd $(INDEXER_PATH); git reset --hard; git pull;)
+	@cd $(INDEXER_PATH) && docker-compose up
+
+indexer-docker-down:
+# stop and remove indexer containers and images
+	@if ! [ -d $(INDEXER_PATH) ]; then	\
+		echo "Repo not found: $(INDEXER_PATH) does not exist."; exit 1; fi
+	@(cd $(INDEXER_PATH); docker-compose down -v --rmi all --remove-orphans;)
+	rm -rf $(INDEXER_PATH)
+
+## --- Indexer jobs (local based)--- ##
+
+setup-postgresql:
+# Source: https://www.postgresql.org/download/linux/ubuntu/
+	sudo sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list'
+	wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -
+	sudo apt-get update; sudo apt-get -y install postgresql;
+# log connection info
+	sudo -u postgres psql -c "\conninfo"
+# Create user (with password) and database. Grant permissions.
+	@echo "Setting up database: User=$(POSTGRES_UNAME) password=$(POSTGRES_PASS) dbname=$(POSTGRES_DBNAME)"
+	@sudo -u postgres createuser $(POSTGRES_UNAME)
+	@sudo -u postgres createdb $(POSTGRES_DBNAME)
+	@sudo -u postgres psql -c "alter user $(POSTGRES_UNAME) with encrypted password '$(POSTGRES_PASS)';"
+	@sudo -u postgres psql -c "grant all privileges on database $(POSTGRES_DBNAME) to $(POSTGRES_UNAME)"
+# List databases
+	@sudo -u postgres psql -c "\l"
```

Comment:
let's also add docker based postgres setup and update the documentation.

Reply:
docker based setup already has the postgres (indexer-db) container: [https://github.com/algorand/indexer/blob/develop/docker-compose.yml#L17](https://github.com/algorand/indexer/blob/develop/docker-compose.yml#L17)

Reply:
updating docs